org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/UnresolvedElementsSubProcessor.java

```diff
@@ -706,6 +717,31 @@ private static void addSimilarTypeProposals(int kind, ICompilationUnit cu, Name
 				}
 			}
 		}
+		if (proposals.size() > 0) {
+			IResource resource = cu.getResource();
+			if (resource != null) {
+				URI uri = resource.getLocationURI();
+				if (uri != null) {
+					if (JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isAddMissingImportsCommandSupport() && JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isAdvancedOrganizeImportsSupported()) {
+						proposals.add(new CUCorrectionCommandProposal(CorrectionMessages.UnresolvedElementsSubProcessor_add_allMissing_imports_description, CodeActionKind.QuickFix, cu, relevance + 10, COMMAND_ID_ACTION_ADDALLMISSINGIMPORTS, Collections.singletonList(uri.toString())));
```

Comment:
Should the "Add all missing imports" proposal appear below the proposal to just add the import from the same line ?

The import from the same line is defined as [`relevance + 100`](https://github.com/eclipse/eclipse.jdt.ls/blob/cf3984c17f2e44df25d67a31df656624e9de4092/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/UnresolvedElementsSubProcessor.java#L731) so it ends up just below it.

![add-all-missing-imports-order](https://user-images.githubusercontent.com/1417342/199624473-874268fc-80ac-4a96-a002-0d2057b86093.png)

Reply:
Do you mean we should assign a new relevance like `relevance + 99` to keep the "add all missing imports" at the end of the specific proposal?

Reply:
I think it's fine for now. Doesn't look like there's any proposal that would go in between. I just wanted to know if you intended for `Add all missing imports` to be below the "Import ... " proposals, but it looks like that's the case.